src/components/Toast/types.ts

```diff
@@ -0,0 +1,22 @@
+import { ReactNode } from "react"
+
+export interface Config {
+  duration: number
+  animationDuration: number
+  position: Position
+  behavior: "stack" | "replace"
+}
+
+export interface Position {
+  vertical: "top" | "center" | "bottom"
+  horizontal: "left" | "center" | "right"
+}
+
+export type Status = "success" | "info" | "error" | "warning"
+
+export type Toast = {
+  id: string
+  message: string
+  status: Status
+  action?: (handleClose: () => void) => ReactNode
```

Comment:
This API (`Toast['action']`) feels maybe a bit rough.

This is an argument that gets passed as a param to a function (addToast), that expects a callback function that will then return some JSX and maybe inject the `handleClose` argument.

I see why you needed this but also wonder if you could have made your life easier by taking a simplistic approach and not taking in custom JSX.

One idea:
```ts
action: {
  type: 'dismiss' | 'dismissAll'
  text: string
}
```

(Or replace action with `dismissText` or `dismissAllText`)

It would let you go from this function call:
```tsx
addToast({
  message: 'Success Toast',
  status: 'success',
  action: (handleClose) => (
    <Button ariaLabel="close toast" onClick={handleClose}>
      Dismiss
    </Button>
  ),
});
```

To this:
```tsx
addToast({
  message: 'Success Toast',
  status: 'success',
  action: {
    text: 'Dismiss',
    type: 'dismiss',
  },
});
```

You lose changeability (all your action buttons would look the same, which could be good for cohesion anyway) but gain a little bit in ergonomics. ⚖️

Reply:
I like this. I agree that this simplifies things.